### PR TITLE
fix: line number for wrapped lines

### PR DESCRIPTION
**Pull request by thorseraq**
Before change, the code block line number is calculated by number of `\n`, when there are long lines, and when they are wrapped, the line number will be more than number of `\n`, it causes line number less than actual lines, as shown below.
image

After, line numbers only cares how much start lines there are, exceeding part are filled up by calculating

https://user-images.githubusercontent.com/20554850/212226891-11c8075d-5a54-4bb5-be89-3197a6979493.mov

### CODE REVIEW

File: packages/blocks/src/code-block/components/syntax-code-block.ts [hunk 1]
```diff
@@ -80,13 +137,15 @@ SyntaxCodeBlock.className = 'ql-syntax';
...
   private _language = 'javascript';
+
   static register() {
     Quill.register(CodeToken, true);
     Quill.register(SyntaxCodeBlock, true);
   }
 
   constructor(quill: QuillType, options: SyntaxCodeBlockOptions) {
     super(quill, options);
+    // console.log('(quill: QuillType, options: SyntaxCodeBlockOptions)');
```
Comment (himself65) [inline packages/blocks/src/code-block/components/syntax-code-block.ts]: remove?

Comment (thorseraq) [reply packages/blocks/src/code-block/components/syntax-code-block.ts]: done

File: packages/blocks/src/code-block/components/syntax-code-block.ts [hunk 2]
```diff
@@ -9,10 +9,25 @@ const Module = Quill.import('core/module');
...
 const CodeToken = Quill.import('modules/syntax');
 
+function addLineNumber(
+  container: HTMLElement,
+  lineHeight: string,
+  i: number | null
+) {
+  const node = document.createElement('div');
+  if (i) {
+    node.innerHTML = `${i}`;
```
Comment (lawvs) [inline packages/blocks/src/code-block/components/syntax-code-block.ts:19]: For safety reasons, use `textContent` please.

**Comment by doodlewind** [comment]
Minor issue found: if one line wraps into more than 2 lines, the count would be incorrect:

image

Context:

```
111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
22222
```

File: packages/blocks/src/code-block/components/syntax-code-block.ts (hunk 2 above)
Comment (thorseraq) [reply packages/blocks/src/code-block/components/syntax-code-block.ts:19]: Thanks, I will change this

**Comment by thorseraq** [comment]
> Minor issue found: if one line wraps into more than 2 lines, the count would be incorrect:
> 
> image
> 
> Content:
> 
> ```
> 111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
> 22222
> ```

It seems the that line length calculated by canvas is not stable across devices, which is used to calculate line numbers of a very long line, I should change to a stable way

**Comment by thorseraq** [comment]
Now uses a fake DOM simulating the code block container to calculate text width, how about try again with

testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
test

**Review by doodlewind — APPROVED** [review]
LGTM

**Pull request merged**